python/ray/data/impl/block_batching.py

```diff
@@ -64,7 +68,19 @@ def batch_block(block: ObjectRef[Block]):
     for block_window in _sliding_window(blocks, prefetch_blocks + 1):
         block_window = list(block_window)
         with stats.iter_wait_s.timer():
-            ray.wait(block_window, num_returns=1, fetch_local=True)
+            context = DatasetContext.get_current()
+            # ray.wait doesn't work as expected so that we use
+            # the actor based prefetcher as a work around. Read
+            # https://github.com/ray-project/ray/issues/23983 for details.
+            if (
+                len(block_window) > 1
+                and context.actor_prefetcher_enabled
+                and not ray.util.client.ray.is_connected()
+            ):
+                prefetcher = get_or_create_prefetcher()
```

Comment:
This will currently execute a named actor handle fetch on ever prefetch window; could we cache this actor handle so it's fetched once per `batch_blocks()` call?